src/sciline/domain.py

```diff
@@ -7,6 +7,10 @@
 
 
 class Scope(Generic[PARAM, SUPER]):
+    """
+    Helper for defining a generic type alias.
+    """
```

Comment:
Should we maybe add examples here...? Including what is `PARAM` and `SUPER`.
It seems like type-argument parameters aren't really shown in the api reference.

Reply:
I am just trying to figure out where to document this. I will most likely add a notebook with examples.

Reply:
... but will probably also wait until we have tried this more in practice, since I suspect the implementation may change a couple more times.